jsettlers.graphics/src/jsettlers/graphics/map/MapContent.java

```diff
@@ -291,27 +301,28 @@ private void drawMessages(GLDrawContext gl) {
 					float bright = color.getRed() + color.getGreen() + color.getBlue();
 					if (bright < .9f) {
 						// black
-						gl.color(1, 1, 1, .5f);
-						gl.fillQuad(x, y, x + width, y + MESSAGE_LINEHIEGHT);
+						drawer.setColor(1, 1, 1, a/2);
 					} else if (bright < 2f) {
 						// bad visibility
-						gl.color(0, 0, 0, .5f);
-						gl.fillQuad(x, y, x + width, y + MESSAGE_LINEHIEGHT);
+						drawer.setColor(0, 0, 0, a/2);
 					}
+					for (int i=-1; i<3; i++)
+						drawer.drawString(x+i, y-1, name);
 					drawer.setColor(color.getRed(), color.getGreen(),
-							color.getBlue(), 1);
+							color.getBlue(), a);
 					drawer.drawString(x, y, name);
 					x += width + 10;
 				}
 
-				drawer.setColor(1, 1, 1, 1);
+				drawer.setColor(1, 1, 1, a);
 				drawer.drawString(x, y, m.getMessage());
 
 				messageIndex++;
-				if (messageIndex >= MAX_MESSAGES) {
+				if (messageIndex >= Messenger.MAX_MESSAGES) {
 					break;
 				}
 			}
+			drawer.setColor(1, 1, 1, 1);
```

Comment:
You don't need to set this. Every component should set the font color before drawing.